tests/Unit/PdfTest.php

```diff
@@ -0,0 +1,262 @@
+                200,
+                fn() => sprintf('image/%s', config('transmorpher.document_default_image_format')),
+            ],
+            'width and height' => [
+                'w-100+h-100',
+                200,
+                fn() => sprintf('image/%s', config('transmorpher.document_default_image_format')),
+            ],
+            'format png' => [
+                'f-png',
+                200,
+                'image/png',
+            ],
+            'format webp' => [
+                'f-webp',
+                200,
+                'image/webp',
+            ],
+            'format jpg' => [
+                'f-jpg',
+                200,
+                'image/jpeg',
+            ],
+            'format gif' => [
+                'f-gif',
+                200,
+                'image/gif',
+            ],
+            'page' => [
+                'p-1',
+                200,
+                fn() => sprintf('image/%s', config('transmorpher.document_default_image_format')),
+            ],
+            'page width height format png' => [
+                'p-1+f-png+w-500+h-1000',
+                200,
+                'image/png',
+            ],
+        ];
+    }
+
+    #[Test]
+    #[Depends('ensurePdfDerivativeCanBeDownloaded')]
+    #[Depends('ensurePdfDerivativeImagesCanBeDownloaded')]
+    public function ensureUnprocessedFilesAreNotAvailable(Version $version)
+    {
+        $version->Media->Versions->each->update(['processed' => 0]);
+
+        $this->get(route('getDocumentDerivative', [$this->user->name, $version->Media]))->assertNotFound();
+
+        return $version;
+    }
+
+    #[Test]
+    #[Depends('ensurePdfDerivativeCanBeDownloaded')]
+    public function ensurePdfMetadataIsRemoved(Version $version)
+    {
+        $config = new PdfParserConfig();
+        $config->setRetainImageContent(false);
+        $pdfParser = new Parser([], $config);
+
+        // Empty values are filtered for easier comparison.
+        $originalMetadata = array_filter($pdfParser->parseFile($this->originalsDisk->path($version->originalFilePath()))->getDetails());
+        $derivativeMetadata = array_filter($pdfParser->parseFile($this->pdfDerivativesDisk->path($version->nonVideoDerivativeFilePath()))->getDetails());
+
+        $metadataComparisonKeys = [
+            'CreationDate',
+            'Subject',
+            'Author',
+            'Creator',
+            'Producer',
+            'ModDate',
+            'Title',
+            'pdf:producer',
+            'xap:creatortool',
+            'xap:modifydate',
+            'xap:createdate',
+            'xap:metadatadate',
+            'dc:title',
+            'dc:creator',
+            'dc:description',
+            'xapmm:documentid',
+            'xapmm:instanceid',
+        ];
+
+        foreach ($metadataComparisonKeys as $key) {
+            array_key_exists($key, $originalMetadata)
+            && array_key_exists($key, $derivativeMetadata)
```

Comment:
maybe just assert that the key is not present or that the content is empty. asserting that it is different isnt safe.

Reply:
also ensure that the stuff is in the test file, else a test file without the metadata would make the test do nothing.

Reply:
I cannot simply look for the key not to be present or empty, as some information will be filled, such as CreationDate, ModDate, Producer, ...

Reply:
I have replaced the PDF with one with more metadata and adjusted the comparison keys.

Reply:
Make metadata removal an option

Reply:
Metadata removal is now optional, I extended the test case.